addon/helpers/log-activity.ts

```diff
@@ -0,0 +1,35 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+import ActivityTracking, { Activity } from '../services/activity-tracking';
+
+export default class extends Helper {
+  @service declare activityTracking: ActivityTracking;
+
+  compute(params: any[]) {
+    const [action, actionDescription] = params;
+    assert(
+      '[helper][log-activity] An action is required for this helper to work properly.',
+      typeof action === 'function'
+    );
+    assert(
+      '[helper][log-activity] An actionDescription needs to be passed for the activity-log to make sense.',
+      typeof actionDescription === 'string'
+    );
+
+    return (event: PointerEvent) => {
+      const activity = this.buildActivityObject(actionDescription);
+      this.activityTracking.log(activity);
+      return action(event);
+    };
+  }
+
+  private buildActivityObject(actionDescription: string): Activity {
+    return {
+      type: 'button_click',
```

Comment:
the helper is called `log-activity` but the type is hard coded as `button_click`. Should we make the name of the helper more context specific? Or the type configurable?

Reply:
For now I don't see any other use case of using the `log-activity` helper except on a click action.
This helper will always be called along with an action, we can totally rename it to be more precise, I'm open to suggestions!
```handlebars
<OSS::Button @label="Open modal" {{on "click" (log-activity this.openModal "Open the product modal")}} />
```

Reply:
We also didn't have any way to define extra params ? (same for others)

Reply:
@Miexil maybe we can handle the params here with named args instead of positional and have something like

```handlebars
(log-activity this.openModal type="foobar" label="Open the product modal")
```

And if no `type` is provided, we can fallback on `button_click` by default?